cmd/runner/rabbit.go

```diff
@@ -28,21 +29,41 @@ var (
 // retrieving and handling StudioML workloads within a self hosted
 // queue context
 
-func initRMQ() (rmq *runner.RabbitMQ) {
-	// NewRabbitMQ takes a URL that has no credentials or tokens attached as the
-	// first parameter and the user name password as the second parameter
-	creds := ""
-	qURL, errGo := url.Parse(os.ExpandEnv(*amqpURL))
-	if errGo != nil {
-		logger.Warn(kv.Wrap(errGo).With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+func rmqConfig() (creds string, queueURL string, mgtURL string, err kv.Error) {
+	if len(*amqpURL) != 0 {
+		qURL, errGo := url.Parse(os.ExpandEnv(*amqpURL))
+		if errGo != nil {
+			logger.Warn(kv.Wrap(errGo).With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		if qURL.User != nil {
+			creds = qURL.User.String()
+		} else {
+			logger.Warn(kv.NewError("missing credentials in url").With("url", *amqpURL).With("stack", stack.Trace().TrimRuntime()).Error())
+		}
+		qURL.User = nil
+		queueURL = qURL.String()
 	}
```

Comment:
Looks like the logic is the same for both URLs, why not factor it out into a function?

Reply:
It felt a little too small to break out.  I would end up with only a small reduction inside the top level function I think.  I could try it and see.

Reply:
It looks OK with the change I'll leave it in